Fix default hostfile location (backported r26440)

https://bugs.gentoo.org/show_bug.cgi?id=420431

Patch by Orion Poplawski <orion AT SPAMFREE cora DOT nwra DOT com>
--- a/orte/runtime/orte_mca_params.c
+++ b/orte/runtime/orte_mca_params.c
@@ -225,7 +225,7 @@ int orte_register_params(void)
                                 false, false, 1000, &orte_timeout_usec_per_proc);
     
     /* default hostfile */
-    asprintf(&orte_default_hostfile, "%s/etc/openmpi-default-hostfile", opal_install_dirs.prefix);
+    asprintf(&orte_default_hostfile, "%s/openmpi-default-hostfile", opal_install_dirs.sysconfdir);
     mca_base_param_reg_string_name("orte", "default_hostfile",
                                    "Name of the default hostfile (relative or absolute path, \"none\" to ignore environmental or default MCA param setting)",
                                    false, false, orte_default_hostfile, &orte_default_hostfile);
